Exchange/handler.ts

```diff
@@ -45,12 +76,47 @@ export const exchangeHandler = (config: IConfig): ExchangeHandlerT => (
     defaultLog.taskEither.errorLeft(
       r => r.detail ?? "Cannot generate Exchange JWT"
     ),
+    TE.chain(jwt =>
+      pipe(
+        decodeToken(jwt),
+        TE.chain(decodedToken =>
+          storeAuditLog(
+            containerClient,
+            {
+              ip: context.req
+                ? (JSON.stringify(
+                    context.req.headers["x-forwarded-client-ip"]
+                  ) as IPString)
+                : ("" as IPString),
```

Comment:
We can relax the ip type to string

```suggestion
             ip: context.req?.headers["x-forwarded-client-ip"] ?? "",
```